### PR TITLE
Allow set cache time on remember function

### PR DESCRIPTION
**Pull request by ChristianGiupponi**
Since the remember method could be used for a lot of different reason and to store different information for a different amount of time the best think is allow user to define the amount of time.

Right not the whole cache decorator use the same amount of time for each store.

I think that in this way we can manage cache with much more flexibility.

### CODE REVIEW

File: Modules/Core/Repositories/Cache/BaseCacheDecorator.php [hunk 1]
```diff
@@ -187,7 +188,12 @@ protected function remember(\Closure $callback, $key = null)
             $store = $store->tags([$this->entityName, 'global']);
         }
 
-        return $store->remember($cacheKey, $this->cacheTime, $callback);
```
Comment (nWidart) [inline Modules/Core/Repositories/Cache/BaseCacheDecorator.php]: Hm since there's a class property, how does it affect other methods?

Comment (nWidart) [reply Modules/Core/Repositories/Cache/BaseCacheDecorator.php]: Ah I see ok, then we could remove the class property?

Comment (ChristianGiupponi) [reply Modules/Core/Repositories/Cache/BaseCacheDecorator.php]: We could, if we remove it we can just initialize it the method, just a sec and send the new PR

Comment (ChristianGiupponi) [reply Modules/Core/Repositories/Cache/BaseCacheDecorator.php]: sent

**Pull request merged**